packages/gatsby-plugin-google-analytics/src/index.js

```diff
@@ -53,4 +53,48 @@ OutboundLink.propTypes = {
   onClick: PropTypes.func,
 }
 
-export { OutboundLink }
+/**
+ * This allows the user to create custom events within their Gatsby projects.
+ * @param {string} category Required - The object that was interacted with (e.g.video)
+ * @param {string} action Required - Type of interaction (e.g. 'play')
+ * @param {string} label Optional - Useful for categorizing events (e.g. 'Spring Campaign')
+ * @param {integer} value Optional - Numeric value associated with the event. (e.g. A product ID)
+ * @param {bool} nonInteraction Optional - If a hit is considered non-interactive.
+ * @param {string} transport Optional - How the events will be sent. The options are 'beacon', 'xhr', or 'image'.
+ * @param {func} hitCallback Optional - Function that gets called as soon as the hit has been successfully sent.
+ *
+ * @link https://developers.google.com/analytics/devguides/collection/analyticsjs/field-reference#events
+ */
+function trackCustomEvent({
+  category,
+  action,
+  label,
+  value,
+  nonInteraction = true,
+  transport,
+  hitCallback,
+}) {
+  if (typeof window !== "undefined" && window.ga) {
+    let trackingEventOptions = {
+      eventCategory: category,
+      eventAction: action,
+      eventLabel: label,
+      eventValue: value,
+      nonInteraction: nonInteraction,
+    }
+
+    if (transport) {
+      trackingEventOptions["transport"] = transport
+    }
+
+    if (hitCallback && typeof hitCallback === "function") {
+      trackingEventOptions["hitCallback"] = function() {
```

Comment:
```suggestion
      trackingEventOptions["hitCallback"] = hitCallback
      
```

I think it can be assigned directly without wrapping in a function?


PS: other idea would be wrapping here in a suggested timeout wrapper function by default:

```js
function createFunctionWithTimeout(callback, opt_timeout) {
  var called = false;
  function fn() {
    if (!called) {
      called = true;
      callback();
    }
  }
  setTimeout(fn, opt_timeout || 1000);
  return fn;
}
```

And:

```suggestion
      trackingEventOptions["hitCallback"] = createFunctionWithTimeout(hitCallback)
      
```

See hint in section hitcallback: https://developers.google.com/analytics/devguides/collection/analyticsjs/sending-hits

Reply:
I like that! Added.